preupg/application.py

```diff
@@ -733,14 +735,15 @@ def run(self):
             retval = self.scan_system()
             if int(retval) != 0:
                 return retval
-            self.summary_report(self.tar_ball_name)
+            retval = self.summary_report(self.tar_ball_name)
             self.common.copy_common_files()
             KickstartGenerator.kickstart_scripts()
             FileHelper.remove_home_issues()
             if self.conf.upload:
-                self.upload_results(self.tar_ball_name)
+                if not self.upload_results(self.tar_ball_name):
+                    retval = ReturnValues.SEND_REPORT_TO_UI
```

Comment:
This is the main change. Now the return value of _upload_results_ is being checked.